framework/src/transfers/MultiAppTransfer.C

```diff
@@ -172,6 +173,66 @@ MultiAppTransfer::checkMultiAppExecuteOn()
                        "execute_on flags"));
 }
 
+void
+MultiAppTransfer::checkPostProcessorExecuteOn(std::string to_check)
+{
+  std::vector<std::shared_ptr<MultiApp>> multi_apps;
+  if (_from_multi_app)
+    multi_apps.push_back(_from_multi_app);
+  if (_to_multi_app)
+    multi_apps.push_back(_to_multi_app);
+  for (std::shared_ptr<MultiApp> multi_app : multi_apps)
+  {
+    // check subapps
+    for (unsigned int i = 0; i < multi_app->numGlobalApps(); i++)
+    {
+      // If we do not have this app, we skip
+      if (!multi_app->hasLocalApp(i))
+        continue;
+      FEProblemBase & sub_problem = multi_app->appProblemBase(i);
+      checkAgainstPPExec(to_check, sub_problem);
+    }
+
+    // check main app
+    FEProblemBase & main_problem = multi_app->problemBase();
+    checkAgainstPPExec(to_check, main_problem);
+  }
+}
+
+void
+MultiAppTransfer::checkAgainstPPExec(std::string to_check, FEProblemBase & problem)
+{
+  // It's possible that the Transfer is going to/from a pp/vpp in a subapp. Therefore, if it doesn't
+  // have the pp/vpp name in this apps problem, return, as this behavior is checked in the subapp
+  // portion of checkPostProcessorExecuteOn.
+  if (!problem.hasUserObject(to_check))
```

Comment:
I would error here
and not call the routine if the problem does not have it

Reply:
but actually just checking this will not work, since there could be a PP with the same name in both problems, and the check would be performed in both directions

Reply:
The check exists to make sure that we are checking against a postprocessor that exists, since it would either be in the subapps or the main app, or both. Since this function gets run once for the main app, and then once for every subapp, if the transfer has pps with the same name in both, it would still check both.

Reply:
but that s a problem because it s likely only one of these two distinct PP with the same name are actually involved in the transfer

Reply:
Are you sure that's even possible? I mean, on face that seems like something you shouldn't be able to do. The transfer can accept a "to_postprocessor" from either the sub or main app, I believe, so how would it even be able to tell the difference between the two identical postprocessors in the first place?

Reply:
I don't see a good way of dealing with this problem other than enforcing unique names for the postprocessors then. Since to_postprocessor and from_postprocessor don't exist in the base MultiAppTransfer class, AFAIK there's no way for MultiAppTransfer to know which is the correct one without doing something like making the function ```checkPostProcessorExecuteOn(std::string to_postprocessor, std::string for_postprocessor) ``` and allowing the parameters to be nullable or something.

Reply:
You're going to have to do something like that.
checkPostProcessorExecuteOn(std::string to_postprocessor, direction)
and call it twice if there's two